frontend/public/components/storage/attach-storage.tsx

```diff
@@ -0,0 +1,468 @@
+        checkboxItems: [
+          {
+            value: 'Mount as read-only',
+          },
+        ],
+        selectedOptions: [],
+      };
+    }
+
+    componentDidMount() {
+      const { kind, metadata } = this.state.resourceObj;
+      const ko = _.find(k8sModels, model => model.plural === kind);
+      const supportedKinds = [
+        'Deployment',
+        'DeploymentConfig',
+        'ReplicaSet',
+        'ReplicationController',
+      ];
+
+      if (!_.includes(supportedKinds, ko.kind)) {
+        this.setState({
+          error: `Storage is not supported for kind ${ko.kind}.`,
+          inProgress: false,
+        });
+        return;
+      }
+
+      // Get the current resource so we can add to its definition
+      k8sGet(ko, metadata.name, metadata.namespace).then(
+        obj => {
+          this.setState({ resourceObj: obj });
+        },
+        err => this.setState({ error: err.message })
+      );
+    }
+
+    onError = error => {
+      this.setState({
+        error,
+        inProgress: false,
+      });
+    };
+
+    handleChange = event => {
+      const name = event.target.name;
+      const value = event.target.value;
+      const formData = { ...this.state.formData, [name]: value };
+      this.setState({ formData });
+    };
+
+    onReadOnlyChanged = event => {
+      let readOnly = this.state.readOnly;
+      readOnly = readOnly ? false : true;
+      this.setState({ readOnly });
+    };
+
+    onFormDisable = disable => {
+      this.setState({
+        disableForm: disable,
+      });
+    };
+
+    isValidForm = () => {
+      // Check that a valid PVC has been choosen
+      if (!this.state.formData.claimName) {
+        this.setState({ error: 'Storage must be selected', inProgress: false });
+        return false;
+      }
+      return true;
+    };
+
+    // Add Attach storage //
+    save = (event: React.FormEvent<EventTarget>) => {
+      event.preventDefault();
+      if (!this.isValidForm()) {
+        return;
+      }
+      const { kind, metadata } = this.state.resourceObj;
+      const ko = _.find(k8sModels, model => model.kind === kind);
+      const resourceObj = { ...this.state.resourceObj };
+      const template = resourceObj.spec.template;
+      let { volumeName, mountPath, subPath, claimName } = this.state.formData;
+      const readOnly = this.state.readOnly;
+      const storage = {
+        attach: null,
+        existingMountPaths: null,
+        volumeAlreadyMounted: null,
+        onPVCSelected: null,
+      };
+      storage.attach = {
+        persistentVolumeClaim: claimName,
+        volumeName: volumeName,
+        mountPath: mountPath,
+        subPath: subPath,
+        readOnly: this.state.readOnly,
+        allContainers: true,
+        containers: {},
+      };
+      const isContainerSelected = function(container) {
+        return (
+          storage.attach.allContainers ||
+          storage.attach.containers[container.name]
+        );
+      };
```

Comment:
```js
const isContainerSelected = ({name}) => storage.attach.allContainers || storage.attach.containers[name];
```

Reply:
You could probably define this above and just use it in `getMountPaths` since the logic doesn't need to change.